subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -165,9 +168,17 @@ private void checkProperties(Properties existingProperties) {
     }
 
     private void testDistributionUrl() {
-        if (distributionUrlConfigured && !isOffline) {
+        if (distributionUrlConfigured) {
             try {
-                new Download(new Logger(true), "gradlew", Download.UNKNOWN_VERSION).sendHeadRequest(getDistributionUrl());
+                String url = getDistributionUrl();
+                URI uri = URI.create(url);
+                if (uri.getScheme().equals("file")) {
+                    if (!Files.exists(Paths.get(uri.getPath()).toAbsolutePath())) {
+                        throw new UncheckedIOException(String.format("Test of distribution url %s failed. Please check the values set with --gradle-distribution-url and --gradle-version.", url));
```

Comment:
:question: Why generate a exception message with the URL in this case and not in the `http` case?

I think we should make the behavior similar for both schemes.